src/code/audio_seqcmd.c

```diff
@@ -0,0 +1,840 @@
+        // print cmd high-bits, "SEQ H"
+        AudioDebug_ScrPrt((const s8*)gAudioDebugTextSeqCmdHighBits, (cmd >> 16) & 0xFFFF);
+        // pring cmd low-bits, "    L"
+        AudioDebug_ScrPrt((const s8*)gAudioDebugTextSeqCmdLowBits, cmd & 0xFFFF);
+    }
+
+    op = cmd >> 28;
+    seqPlayerIndex = (cmd & 0xF000000) >> 24;
+
+    switch (op) {
+        case SEQCMD_OP_PLAY_SEQUENCE:
+            // Play a new sequence
+            seqId = cmd & 0xFF;
+            seqArgs = (cmd & 0xFF00) >> 8;
+            // `fadeTimer` is only shifted 13 bytes instead of 16 bytes.
+            // `fadeTimer` continues to be scaled in `Audio_StartSequence`
+            fadeTimer = (cmd & 0xFF0000) >> 13;
+            if (!gActiveSeqs[seqPlayerIndex].isWaitingForFonts && (seqArgs < 0x80)) {
+                Audio_StartSequence(seqPlayerIndex, seqId, seqArgs, fadeTimer);
+            }
+            break;
+
+        case SEQCMD_OP_STOP_SEQUENCE:
+            // Stop a sequence and disable the sequence player
+            fadeTimer = (cmd & 0xFF0000) >> 13;
+            Audio_StopSequenceNow(seqPlayerIndex, fadeTimer);
+            break;
+
+        case SEQCMD_OP_QUEUE_SEQUENCE:
+            // Queue a sequence into `sSeqRequests`
+            seqId = cmd & 0xFF;
+            seqArgs = (cmd & 0xFF00) >> 8;
+            fadeTimer = (cmd & 0xFF0000) >> 13;
+            priority = seqArgs;
+
+            // Checks if the requested sequence is first in the list of requests
+            // If it is already queued and first in the list, then play the sequence immediately
+            for (i = 0; i < sNumSeqRequests[seqPlayerIndex]; i++) {
+                if (sSeqRequests[seqPlayerIndex][i].seqId == seqId) {
+                    if (i == 0) {
+                        Audio_StartSequence(seqPlayerIndex, seqId, seqArgs, fadeTimer);
+                    }
+                    return;
+                }
+            }
+
+            // Searches the sequence requests for the first request that does not have a higher priority
+            // than the current incoming request
+            found = sNumSeqRequests[seqPlayerIndex];
+            for (i = 0; i < sNumSeqRequests[seqPlayerIndex]; i++) {
+                if (priority >= sSeqRequests[seqPlayerIndex][i].priority) {
+                    found = i;
+                    i = sNumSeqRequests[seqPlayerIndex]; // "break;"
+                }
+            }
+
+            // Check if the queue is full
+            if (sNumSeqRequests[seqPlayerIndex] < 5) {
+                sNumSeqRequests[seqPlayerIndex]++;
+            }
+
+            for (i = sNumSeqRequests[seqPlayerIndex] - 1; i != found; i--) {
+                // Move all requests of lower priority backwards 1 place in the queue
+                sSeqRequests[seqPlayerIndex][i].priority = sSeqRequests[seqPlayerIndex][i - 1].priority;
+                sSeqRequests[seqPlayerIndex][i].seqId = sSeqRequests[seqPlayerIndex][i - 1].seqId;
```

Comment:
may be worth commenting on the side effect that if the queue is full the lowest priority request is dropped (or overwritten by the incoming request if that incoming one is an even lower priority)